src/garage/experiment/local_tf_runner.py

```diff
@@ -241,8 +252,11 @@ def restore(self, snapshot_dir, from_epoch='last'):
             specify manually or through run_experiment() interface.
 
         """
-        snapshotter.snapshot_dir = snapshot_dir
-        saved = snapshotter.load(from_epoch)
+        assert self.snapshotter is not None, (
+            'You must specify SnapshotConfig in LocalRunner construction '
```

Comment:
this should be an exception, not an assert.

i think a more apt error is "You cannot restore this LocalRunner because a snapshot is required to do so. You can ensure your experiment produces a snapshot directory by providing a SnapshotConfig to LocalRunner during training."

with that said, I think a lot of your code could be simpler if you just disallowed the no-snapshots case, and I recommend you do that. Snapshots don't take up much space, especially if your snapshot_mode is 'last'.